### PR TITLE
Config lockfile

### PR DESCRIPTION
**Pull request by MattF-NSIDC**

### CODE REVIEW

File: qgreenland/models/base_model.py [hunk 1]
```diff
@@ -20,3 +21,8 @@ class Config:
         # https://github.com/samuelcolvin/pydantic/issues/1241
         # https://github.com/samuelcolvin/pydantic/issues/2763
         keep_untouched = (cached_property,)
+
+        arbitrary_types_allowed = True
+
+    def __json__(self) -> dict[Any, Any]:
+        return self.dict()
```
Comment (MattF-NSIDC) [inline qgreenland/models/base_model.py:28]: Also define `__json__` for layer and dataset models? Currently full datasets are being repeated throughout the config, including all of their assets.

**Comment by MattF-NSIDC** [comment]
TODO: Use relative paths for repository assets

File: doc/CONTRIBUTING.md [hunk 2]
```diff
@@ -9,6 +9,18 @@ release except releases labeled as "stable". Stable releases can be found at
...
 
 
+## NEW NEW configuration
+
+TODO!!! Cleanup old configuration sections.
+
+
+### Lockfile
+
+Use `inv export > qgreenland/config/cfg-lock.json` to refresh this lockfile.
```
Comment (MattF-NSIDC) [inline doc/CONTRIBUTING.md:19]: TODO: Change to `inv config.export`

**Comment by MattF-NSIDC** [comment]
TODO: Remove `datasets` and `layers` config keys. They're already represented in `layer_tree`.

File: qgreenland/models/base_model.py (hunk 1 above)
Comment (hopesimonoko) [reply qgreenland/models/base_model.py:28]: would this look like def __json__(self) -> Layer[...]...?

Comment (MattF-NSIDC) [reply qgreenland/models/base_model.py:28]: Resolved this by specifying `__json__` for the `Config` model. Excludes the `layers` and `datasets` elements; only the tree matters.

**Pull request merged**